### PR TITLE
Including initial project documentation

### LINKED ISSUE #17: Create project documentation

### PR DESCRIPTION
**Pull request by alexfrt**
Closes #17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,21 @@
-# mesh-connect
\ No newline at end of file
+# Mesh Connect
+Mesh Connect is a project that aims to deliver a low cost communication infrastructure to areas that have no Internet connectivity at all. The project relies on low cost ESP32 devices with WiFi and LoRa radios. The role of WiFi connectivity is to allow any smartphone or PC to consume the Internet service provided by the ESP32 devices, which will form a mesh network through LoRa, being capable of deliver data over long distances with no need of a expensive infrastructure. Naturally, LoRa radios does not provide a large network throughput, but it is enought to allow basic texting applications to be used by the end users that live in areas with no Internet connectity at all.
```
Comment (pxds) [inline README.md:2]: Change "being capable of deliver" for just "delivering"

Comment (pxds) [reply README.md:2]: "need of a" change "a" for "an"

Comment (pxds) [reply README.md:2]: "Naturally, LoRa radios does" change "does" for "do"

Comment (pxds) [reply README.md:2]: "enought" doesn't have that t at the end, change for "enough"

File: README.md [hunk 2]
```diff
@@ -1 +1,21 @@
...
+
+## Topology overview
+Each ESP32 that joins the Mesh Connect network topology may assume one of two roles:
```
Comment (pxds) [inline README.md:5]: change "may" for "can" because we're talking about something that will definetely happen

File: README.md [hunk 3]
```diff
@@ -1 +1,21 @@
...
+- **Node**, a device that will become an WiFi Access Point that provides Internet connectivity to any device within the range. Nodes send any incoming TCP/IP traffic to Gateway nodes through the LoRa mesh.
+- **Gateway**, a device that will connect to an existing WiFi Access Point that is somehow connected to the Internet. It receives traffic from other Nodes through the LoRa mesh and route it to the Internet using its WiFi connection. 
+
+![Topology Overview](docs/topology.png)
+
+### How do devices become nodes or gateways?
+When the device starts, it scans the WiFi SSIDs available in its range. If there is a network named as *MeshConnectGWAP* with no authentication required, it will automatically try to become a WiFi client of such network, thus becoming a Gateway. Otherwise it will become a node and establish communication with peering nodes through the LoRa mesh network.
```
Comment (pxds) [inline README.md:12]: "it scans the WiFi SSIDs available in its range" change for "it scans if there are any WiFi SSIDs available in its range"

**Review by pxds — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (alexfrt) [reply README.md:12]: It is going to scan, **unconditionally**.

**Review by pxds — APPROVED** [review]

**Pull request merged**